addons/purchase_stock/wizard/product_replenish.py

```diff
@@ -51,6 +51,7 @@ def _prepare_run_values(self):
         res = super()._prepare_run_values()
         if self.supplier_id:
             res['supplierinfo_id'] = self.supplier_id
```

Comment:
Here we fill `supplierinfo_id` but get `supplierinfo_name` in `run_buy`

Reply:
first we create procurement and that procurement has supplier_id set 
that procurement will create a move
and the move will create a new procurement for run_buy and there supplier is not set anywhere

Reply:
this why I'm setting the partner on the group_id
alternatively we can keep the supplierinfo in the context

Reply:
@HANNICHE-Walid It seems your test is working without your patch. `supplierinfo_id` filled in values is correctly retrieved in `_run_buy`

Reply:
It's because the reception should be done in more than 1 step apparently